pommerman/forward_model.py

```diff
@@ -499,14 +499,30 @@ def make_bomb_maps(position):
             ''' Makes an array of an agents bombs and the bombs attributes '''
             blast_strengths = np.zeros((board_size, board_size))
             life = np.zeros((board_size, board_size))
+            moving_direction = np.zeros((board_size, board_size))
 
             for bomb in bombs:
                 x, y = bomb.position
                 if not is_partially_observable \
                    or in_view_range(position, x, y):
                     blast_strengths[(x, y)] = bomb.blast_strength
                     life[(x, y)] = bomb.life
-            return blast_strengths, life
+                    if bomb.moving_direction is not None:
+                        moving_direction[(x, y)] = bomb.moving_direction.value
+                    else:
+                        moving_direction[(x, y)] = 0
+            return blast_strengths, life, moving_direction
+
+        def make_flame_map(position):
+            ''' Makes an array of an agents flame life'''
+            life = np.zeros((board_size, board_size))
+
+            for flame in flames:
+                x, y = flame.position
+                if not is_partially_observable \
+                   or in_view_range(position, x, y):
+                    life[(x, y)] = flame.life + 1
```

Comment:
why +1?

Reply:
```python
for flame in curr_flames:
    position = flame.position
    if flame.is_dead():
        item_value = curr_items.get(position)
        if item_value:
            del curr_items[position]
        else:
            item_value = constants.Item.Passage.value
        curr_board[position] = item_value
    else:
        flame.tick()
        flames.append(flame)
```
Because the flame ticking starts at 2, then ticks down every step and is deleted one step **after** the life reaches 0. Therefore the flame life ticks down from 2 to 1 to 0, but at tick 0 it is still in the environment for one more step. So to accurately represent the flame life it needs to tick down 3, 2, 1, 0. Or we change the code to initialize flames with life 3 and put `flame.tick()` above `if flame.is_dead():`

Reply:
👍 let's stick with this. thanks for clarifying. want to add a brief comment explaining that so it's not confusing later?

Reply:
Sure thing, give me a minute